docs/concepts.md

```diff
@@ -0,0 +1,153 @@
+# Concepts
+
+PrairieLearn is extremely flexible and powerful, but this flexibility can appear opaque at first. This document provides an overview of the key concepts in PrairieLearn, and how they fit together.
+
+## High-level overview
+
+A [**course**](./course.md) in PrairieLearn is composed of **course instances** and **questions**. Each [course instance](./courseInstance.md) is an offering of a course in a particular semester/term. For example, a course "CS 225" might have instances "CS 225 Spring 2022" and "CS 225 Fall 2022". A course instance contains all the **assessments** for that particular term. The course itself contains the questions that are shared across all instances. You can think of the course as containing a bank of questions that are then used to create assessments for each instance of the course, and a way to organize every instance of the course.
+
+=== "Concept Map"
+
+    ```d2
+    --8<-- "docs/diagrams/concept-map.d2"
+    ```
+
+=== "Example for the `PrairieLearn 101` course"
```

Comment:
Did a slightly different layout, hopefully a little more obvious now. Still hidden as the image is quite large.

Reply:
That's now very clear!

However, I'm wondering whether we'd be better off scrapping both of the existing diagrams and instead having just one diagram which is like the current example, but with one two questions?

My thinking is that most users understand concrete examples better than abstract ideas. In the concrete example, perhaps we could use labels like "Course Instance: Fall 2024" or "Fall 2024 (course instance)" and similarly for the other objects. At the moment, a user needs to map between the abstract and concrete diagrams to figure out that green boxes are assessments, for example. I think this might be clearer with just one diagram that labeled the green boxes with "HW 1 (assessment)" or similar. I also think that we don't need so many questions in the example. Maybe just two questions, one with a single MC element, the second with numeric and custom diagram elements? And then Exam 1 could include both questions, and HW 1 could just include the second question? I feel like that makes the cardinality of the relationships clear?

I'm also wondering whether we can skip the assessment set layer for this diagram? I personally tend to think of assessments as living directly within course instances, not nested inside assessment sets. Assessment sets themselves are more like an additional "type" property, similar to "module". Also assessment sets are actually children of the course, not course instance, strictly speaking. Just as we don't include modules on this diagram, I think we could avoid assessment sets as well, and then later we have the explanation of both assessment sets and modules together in the text.

Reply:
Maybe we can discuss this on Slack, I am open to changing it.